attach-to-chat 1: components and stories

songgao opened the pull request:
This is the v2 of https://github.com/keybase/client/pull/15320 where we decided to have a new design for the conversation selector.

@cecileboucheron How does this look? Let me know how I should tweak it!

cc @mmaxim @jzila 

![image](https://user-images.githubusercontent.com/255797/50877204-af0db500-1385-11e9-8837-49807e01b90a.png)

![image](https://user-images.githubusercontent.com/255797/50877212-ba60e080-1385-11e9-8585-e1f1b41f71f4.png)

![image](https://user-images.githubusercontent.com/255797/50877217-bfbe2b00-1385-11e9-8f0c-ac3357cb1e10.png)

depends on https://github.com/keybase/client/pull/15383 and https://github.com/keybase/client/pull/15384

chrisnojima commented on shared/chat/conversation-list/choose-conversation.native.js:
```diff
@@ -0,0 +1,12 @@
+export default (props: Props) => (
+  <Kb.Box2 direction="horizontal" centerChildren={true} style={{padding: 32}}>
+    <Kb.Text type="BodySmallError">
+      choose-conversation is not supported on mobile. If you see this outside storybook something is wrong.
```
why even have this? just use choose-conversation.desktop (it works on both) and just never route here

songgao replied:
I was just thinking if someone makes a mistake and uses this on native, this'd make it very clear that it shouldn't happen that way.

chrisnojima replied:
but then we need a .js.flow and a .native that does nothing. lets just not have it

chrisnojima commented on shared/chat/conversation-list/conversation-list-container.js:
```diff
@@ -0,0 +1,20 @@
+import {namedConnect} from '../../util/container'
+import ConversationList from './conversation-list'
+
+type OwnProps = {
```
all these prop types should be exact

chrisnojima commented on shared/chat/conversation-list/conversation-list-container.js:
```diff
@@ -0,0 +1,20 @@
+const mapStateToProps = state => ({})
+const mapDispatchToProps = dispatch => ({})
+const mergeProps = (stateProps, dispatchProps, ownProps) => ({
+  rows: [],
```
???

songgao replied:
Just a placeholder; Will be populated in a follow-up PRs

chrisnojima commented on shared/chat/inbox/row/big-team-channel/container.js:
```diff
@@ -5,10 +5,10 @@ import * as Chat2Gen from '../../../../actions/chat2-gen'
 import {connect, isMobile} from '../../../../util/container'
 
-type OwnProps = {|
+type OwnProps = {
```
???

chrisnojima commented on shared/chat/selectable-big-team-channel-container.js:
```diff
@@ -4,7 +4,7 @@ import SelectableBigTeamChannel from './selectable-big-team-channel'
 import {namedConnect} from '../util/container'
 
-type OwnProps = {|
+export type OwnProps = {|
```
why do we need this

chrisnojima commented on shared/chat/selectable-big-team-channel.js:
```diff
@@ -73,29 +83,34 @@ export const rowHeight = isMobile ? 64 : 56
-    common: {flexBasis: '70%'},
+    common: {
+      flexBasis: '70%',
+      flexShrink: 0.3,
```
why this change

songgao replied:
flexBasis didn't work well. On desktop, if the team name is too long, then we'd end up having channel name taking 70% space that has a bunch of white space. On mobile if team name is too long, we'd have channel name pushed to far right. Looks like flexShrink doesn't work perfectly either, but I'll play with it and see

songgao replied:
OK I think I've found a way to make it work, by using maxWidth and flexShrink. The result is that channelname only grows if it's overflowing, but only up to 70%. If within 70% channelname can show in full, we just shrink teamname until they fit. If both needs shrinking, then it's 3:7 between teamname and channel name. Here are some screenshots:

![image](https://user-images.githubusercontent.com/255797/51571388-e6d12e00-1e56-11e9-959e-6cfaef9c06a4.png)

![image](https://user-images.githubusercontent.com/255797/51571398-eb95e200-1e56-11e9-8d96-fc26b040f9b1.png)

chrisnojima commented on shared/fs/send-attachment-to-chat/index.desktop.js:
```diff
@@ -0,0 +1,66 @@
+import ChooseConversation from '../../chat/conversation-list/choose-conversation'
+import type {Props} from '.'
+
+const SendAttachmentToChat = (props: Props) => (
```
if we can have these in one file vs 2 i think we should. there's nothing platform specific about the implementation that requires a different file so I think its better to just have index.js and use Styles.isMobile inside

songgao replied:
Sounds good; I'll merge them in the next PR.

songgao commented:
@chrisnojima I addressed all feedback except merging the platform files (which I'm gonna do in next PR), and the CSS changes for the big team row (where I made more changes). Ready for another look; thanks!

songgao commented on shared/react-native/ios/Keybase/Storybook.m:
```diff
@@ -24,7 +24,7 @@ + (BOOL)requiresMainQueueSetup
 {
   // Set this to true to enable storybook mode
-  return @{@"isStorybook": @false};
+  return @{@"isStorybook": @true};
```
oops (reminder to myself to revert this before merging)

songgao commented:
bump!
